docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc

```diff
@@ -0,0 +1,207 @@
+
+{ProductName} can build applications that are written in the following language stacks and frameworks:
+
+[cols=2*,options="header"]
+|===
+|Language
+|Framework
+
+|Java
+|Spring Boot, Quarkus
+
+|JavaScript
+|Node.js, Express
+
+|Python
+|Flask
+
+|Go
+|N/A
+|===
+
+//// 
+From JSese: May 15, 2023, Commenting out per Matt Reid and Christian V. Support for private repos is post-Summit.
+[discrete]
+== Granting App Studio access to your Git repository
+
+If {ProductName} can’t access your Git repository, the following message displays: “We can’t access your repository, probably because it’s either private or the URL is incorrect.” If your repository is private, choose one of the following two ways to allow {ProductName} to interact with your repository:
+
+[discrete]
+=== Granting App Studio access to your Git repository through GitHub OAuth
+
+. Under **Authorization**, click **Sign in**. **GitHub Authorize OAuth** opens in a new tab.
+. Click **Authorize _redhat-appstudio_**.
+
+[discrete]
+=== Granting App Studio access to your Git repository with a token
+
+. Under **Authorization**, click **Use a token instead**. The **Authenticate with API token** modal opens.
+. Enter your **Username**.
+. Enter an **API token**.
+. Click **Connect**.
+////
+
+//[id="proc_reviewing_configging_components_{context}"]
+
+== Resolving problems with importing code
+
+[role="_abstract"]
+When you import code, {ProductName} analyzes your repository to determine the right runtime to use to build and deploy your application. In some cases, however, {ProductName} is unable to find the right runtime, and this can cause your code import to fail. These are the most common causes of issues with importing code to {ProductName}:
+
+* **Your repository contains a Dockerfile in an unexpected directory.**
++
+If {ProductName} finds a Dockerfile in your repository, it determines that a Dockerfile runtime is the most suitable for building and deploying you application. However, if the Dockerfile is not saved in one of the following expected locations, an error in determining the right runtime for your components can occur:
++
+[#expected Dockerfile locations]
+.Expected Dockerfile locations
+** Dockerfile
+** docker/Dockerfile
+** .docker/Dockerfile or build/Dockerfile
+** Containerfile
+** docker/Containerfile
+** .docker/Containerfile or build/Containerfile
+
+**Solution:** On the **Review and configure for deployment** page, make sure **Runtime** is set to **Dockerfile**, then verify that you saved your Dockerfile in one of the expected directories.
+
+[TIP]
+====
+Test your Dockerfile in a local environment first to make sure it's valid.
+====
+
+[#repository requirements]
+* **Your repository is missing a Dockerfile or devfile.**
```

Comment:
`missing` sounds like the user needs to have it. Should we say something like "Your repository does not contain a Dockerfile nor devfile"?